firebase-installations/src/main/java/com/google/firebase/installations/CrossProcessLock.java

```diff
@@ -64,13 +64,15 @@ static CrossProcessLock acquire(Context appContext, String lockName) {
           lock.release();
         } catch (IOException e2) {
           // nothing to do here
+          Log.e(TAG, "encountered error while releasing the lock, ignoring", e);
```

Comment:
@vkryachko WDYT about these exceptions we ignore? Is it worth propagating them? Even if we throw them and that results in app crashes - Is there anything we can do about it?

Reply:
I think igrnoring the exceptions should be fine, **but** one thing I found is that there seems to be a resource leak above, i.e. you never `close()` the `RandomAccessFile` which leads to its `fd` to not be release in a timely manner. I wonder if that could cause the file lock not to release properly?

In general I suggest you rewrite all `RandomAccessFile`, `FileChannel` and `FileLock` as one try-with-resources block(it's fully supported at this point).

Reply:
Thanks for the suggestion. I changed the code to close RandomAccessFile in finally block.

Reason, we want the FileChannel and FileLock to be closed later by the caller method acquiring the lock. PTAL.

Reply:
> Reason, we want the FileChannel and FileLock to be closed later by the caller method acquiring the lock. PTAL.

That makes sense, although I would suggest(optional) that you refactor it to make it possible to use `try-with-resources`, e.g. something like this should work:

```
return CrossProcessLock.whileLocked(context, "name", () -> {
  return persistedInstallation.insertOrUpdatePersistedInstallationEntry(prefs);
});
```

this makes sure locking/unlocking is handled inside `whileLocked` as opposed to being up to the caller of `lock()` which could lead to inconsistent behavior and lead to subtle bugs.

Reply:
PTAL. I tried incorporating the changes you recommened.